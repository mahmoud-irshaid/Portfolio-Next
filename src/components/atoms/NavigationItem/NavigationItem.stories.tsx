/* eslint-disable import/no-anonymous-default-export */
import { NavigationItem, NavigationItemProps } from './NavigationItem';

export default {
	title: 'atoms/NavigationItem',
	component: NavigationItem,
};

export const NavigationItemStory = (args: NavigationItemProps) => (
	<ul className="flex gap-8 text-lg">
		<NavigationItem {...args} />
	</ul>
);

NavigationItemStory.storyName = 'NavigationItem';
NavigationItemStory.args = {
	href: '/',
	title: 'Navigation Link',
};
