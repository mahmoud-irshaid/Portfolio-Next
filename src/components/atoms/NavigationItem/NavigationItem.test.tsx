import { NavigationItem } from '../NavigationItem';

import { render } from '@testing-library/react';
import React from 'react';

jest.mock('next/navigation', () => ({
	__esModule: true,
	usePathname: jest.fn().mockReturnValue('/'),
}));

describe('NavigationItem', () => {
	it('renders correctly', () => {
		const { container } = render(
			<NavigationItem
				href="/"
				title="Navigation Link"
				variants={{
					hidden: { opacity: 0, x: -50 },
					visible: (custom: number) => ({
						opacity: 1,
						x: 0,
						transition: { delay: custom },
					}),
				}}
				initial="initial"
				animate="animate"
				customDelay={0}
			/>
		);
		expect(container).toMatchSnapshot();
	});
});
