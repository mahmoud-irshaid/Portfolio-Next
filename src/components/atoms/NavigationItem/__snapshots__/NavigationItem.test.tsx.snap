// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItem renders correctly 1`] = `
<div>
  <li>
    <a
      class="font-bold text-off-black dark:text-off-white md:underlined relative block whitespace-nowrap text-2xl transition md:text-lg"
      href="/"
    >
      Navigation Link
    </a>
  </li>
</div>
`;
